### PR TITLE
docs: Add namespace info to create source

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation

Add docs

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: doc/user/content/sql/create-source/_index.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ menu:
 ---
 
 A [source](../../overview/key-concepts/#sources) describes an external system you want Materialize to read data from, and provides details about how to decode and interpret that data. To create a source, you must specify a [connector](#connectors), a [format](#formats) and an [envelope](#envelopes).
+Like other relations, sources are namespaced by a database and schema.
```
Comment (jseldess) [inline doc/user/content/sql/create-source/_index.md:15]: nit: Might be nice to make `namespaced` a link to https://materialize.com/docs/sql/namespaces/.

**Review by jseldess — APPROVED** [review]
LGTM, with a nit.

Also, can you add this language to the other, more specific `create source` pages in https://github.com/MaterializeInc/materialize/tree/main/doc/user/content/sql/create-source?

**Comment by jkosh44** [comment]
> Also, can you add this language to the other, more specific `create source` pages in https://github.com/MaterializeInc/materialize/tree/main/doc/user/content/sql/create-source?

Just to clarify, you want me to add this same line to the `kafka.md`, `load-generator.md`, `materialize-cdc.md` and `postgres.md` pages?

**Comment by jseldess** [comment]
> Just to clarify, you want me to add this same line to the kafka.md, load-generator.md, materialize-cdc.md and postgres.md pages?

Yeah, I'd add this detail to those pages as well. I agree with Marta that we should improve the namespacing page (opened an issue: https://github.com/MaterializeInc/materialize/issues/19033), perhaps especially given our upcoming RBAC release, which depends on a clear sense of the object hierarchy. But even with that page improved, users should get this understanding more immediately when looking at the source commands, as its not going to be intuitive to users that sources are an object like "tables and views". Thus, Arjun's confusion that triggered this change.

**Comment by morsapaes** [comment]
> users should get this understanding more immediately when looking at the source commands, as its not going to be intuitive to users that sources are an object like "tables and views". Thus, Arjun's confusion that triggered this change.

Alternative take: `SHOW SOURCES`/`SHOW SINKS` should return the database and schema. We are [presenting objects in the UI with the fully qualified name](https://app.campsite.design/materialize/posts/e1n0fsqkkkty), so this could reduce confusion without bloating the source/sink docs with (IMO) unnecessary details.

**Comment by jkosh44** [comment]
This seems like it's turning into a bigger docs change than just a couple of lines, so I'm going to step away and let @MaterializeInc/docs take over. I think any changes that we make to the docs for sources regarding name-spacing and any change we make to the output of `SHOW SOURCES` should also be made for tables, sinks, views, materialized views, indexes, types, functions, secrets, and connections.

**Comment by morsapaes** [comment]
Yeah, I meant it for `SHOW` for any database objects. Opened #19037 with the proposal (verbose as it might be).

**Comment by jkosh44** [comment]
As per an in-person discussion with @jseldess, I'm going to merge this as is and let @MaterializeInc/docs make any additional changes to other object types.

**Pull request merged**